### PR TITLE
fix: use independent response_mask in rollout result. attention_mask used only as valid tokens

### PR DESCRIPTION
**Pull request by qurakchin**
### Description

RLinf uses attention_mask to represent all valid prompt_ids and response_ids, but when using agent, we need to mask the tool response ids in adv function and loss function. In last version I used attention_mask and set tool response ids to 0 in attention_mask, but some code use attention_mask.sum() to calc sequence length, so it's wrong in agent.

In this version, I added a response_mask to represent llm generated response_ids, and attention_mask still represent all valid prompt_ids and response_ids, so it's ok to use attention_mask.sum() to calc sequence length, and in adv function and loss function, i will try to use response_mask to filter tool response ids. Then the bug is fixed.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by qurakchin** [comment]
> What is the purpose of this PR? Could you add some descriptions in the PR description?

Thanks for the reminder. Added.

File: rlinf/data/io_struct.py [hunk 1]
```diff
@@ -305,6 +283,39 @@ def _get_attention_masks_and_position_ids(
...
         return attention_mask, position_ids
 
+    @staticmethod
+    def _get_response_masks(
+        response_mask: list[list[int]],
+        prompt_lengths: torch.Tensor,
+        max_prompt_len: int,
+        total_len: int,
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        max_response_len = total_len - max_prompt_len
```
Comment (Lin-xs) [inline rlinf/data/io_struct.py]: add doc

File: rlinf/data/io_struct.py [hunk 2]
```diff
@@ -305,6 +283,39 @@ def _get_attention_masks_and_position_ids(
...
+        # Broadcast [B, total_len]
+        prompt_start = prompt_start.unsqueeze(1)
+
+        response_mask = batch_pad_to_fixed_len(
+            [torch.as_tensor(ids, dtype=torch.long) for ids in response_mask],
+            max_batch_len=max_response_len,
+            pad_token=0,
+        )
+
+        response_mask = torch.cat(
```
Comment (Lin-xs) [inline rlinf/data/io_struct.py]: delete prompt mask

File: rlinf/data/io_struct.py [hunk 3]
```diff
@@ -740,7 +755,7 @@ def to_actor_batch(
                 shape ``[batch_size, training_seq_length - data_seq_length]``.
         """
 
-        # len = training_seq_length: input_ids, attention_mask, position_ids
+        # len = training_seq_length: input_ids, attention_mask, position_ids, response_mask
```
Comment (Lin-xs) [inline rlinf/data/io_struct.py:752]: add comments for response_mask

Comment (qurakchin) [reply rlinf/data/io_struct.py:752]: added.

File: rlinf/data/io_struct.py (hunk 2 above)
Comment (qurakchin) [reply rlinf/data/io_struct.py]: now the shape is same as attention_mask. should not delete prompt part now.

File: rlinf/data/io_struct.py (hunk 1 above)
Comment (qurakchin) [reply rlinf/data/io_struct.py]: added.
